vignettes/icommands.Rmd

```diff
@@ -0,0 +1,203 @@
+
+Next to these functions, `{rirods}` provides two functions to store R objects directly into iRODS or read RDS files directly from iRODS: `isaveRDS()` and `ireadRDS()`. There is no counterpart in iCommands. To achieve the same effect as `isaveRSD()`, a user would first have to save their object with `saveRDS()`, then send it to iRODS with `iput` and finally remove it with `irm -f`; in the same vein, the effect of `ireadRDS()` would be achieved by retrieving the file with `iget`, reading it with `readRDS()` and then removing it from the local space.
+Read more about this in `vignette("local-irods")`.
+
+## Metadata and querying
+
+The `{rirods}` `imeta()` function covers two of the main functionalities of the `imeta` iCommand: adding and removing metadata from a data object or collection. This function has three main arguments: the logical path of the data object or collection on which to add the metadata, the entity type (e.g. "data_object", the default, or "collection") and a list of operations, in which it is specified whether a certain AVU should be added or removed.
+
+```{r}
+icd("some_collection")
+ils(metadata=TRUE)
+
+# imeta add -C subcollection foo bar baz
+imeta("subcollection", "collection",
+      operations = list(
+        list(operation="add", attribute="foo", value="bar", units="baz")
+        )
+      )
+
+# imeta rm -d random_numbers.rds distribution normal
+imeta("random_numbers.rds",
+      operations = list(
+        list(operation="remove", attribute="distribution", value="normal")
+      ))
+
+ils(metadata=TRUE)
+```
+
+In order to "modify" an AVU with `{rirods}`, we would have to remove it and add its replacement. This also illustrates how we can add/remove several AVUs in one `imeta()` call, which would not be possible with `iCommands`.
+
+```{r}
+# imeta mod -d random_numbers.rds length 100 items u:elements
+imeta("random_numbers.rds",
+      operations = list(
+        list(operation="remove", attribute="length", value="100", units="items"),
+        list(operation="add", attribute="length", value="100", units="elements")
+      ))
+ils(metadata=TRUE)
+```
+
+As mentioned before, whereas the iCommand `imeta ls` is used to list the metadata of an item, `ils(metadata=TRUE)` is used in `{rirods}` instead.
+
+Finally, the `{rirods}` equivalent of iCommand `iquest` is `iquery()`, with the same GenQuery expression as main argument and a few compatible arguments. Given a `query` variable such as "SELECT DATA_NAME, DATA_CHECKSUM WHERE COLL_NAME LIKE '/tempZone/home/rods%'":
+
+`{rirods}` | `iCommands`
+----- | -----
+`iquery(query)` | `iquest "$query"`
+`iquery(query, case_sensitive = FALSE)` | `iquest uppercase "$query"`
+`iquery(query, distinct = FALSE)` | `iquest no-distinct "$query"`
+
+To learn more about `imeta()` and `iquery()`, see `vignette("metadata")`.
+
+```{r cleanup, include=FALSE}
+# httptest2::end_vignette()
```

Comment:
Here and at other places. Do the `http2test` snapshots not work?

Reply:
Fixed!